### PR TITLE
Fix #260

### LINKED ISSUE #260: HeavySelect2Widget is not rendering selected options after ValidationError is raised on form
The problem is traced to [`render_options`](https://github.com/applegrew/django-select2/blob/master/django_select2/forms.py#L230) call where `selected_choices` (second element in `args` list) is list of `unicode`s instead of expected list of integers. `Django==1.8.9`

### PR DESCRIPTION
**Pull request by darbula**
Fix for #260.

### CODE REVIEW

**Comment by codingjoe** [comment]
Since I didn't get the ticket, please add tests, that exploit the bug.

**Comment by darbula** [comment]
I have explained error in more detail in the issue. I would be glad to write some tests if the error is acknowledged by you.

**Comment by codingjoe** [comment]
Sounds legit, please write a test and it should be fine.

File: django_select2/forms.py [hunk 1]
```diff
@@ -408,8 +409,7 @@ def render_options(self, *args):
             choices = {(obj.pk, self.label_from_instance(obj))
                        for obj in self.choices.queryset.filter(pk__in=selected_choices)}
         else:
-            choices = {(k, v) for k, v in choices if k in selected_choices}
-        selected_choices = {force_text(v) for v in selected_choices}
+            choices = {(k, v) for k, v in choices if force_text(k) in selected_choices}
```
Comment (codingjoe) [inline django_select2/forms.py]: shouldn't it be `self.label_form_instance(k)`?

**Comment by codingjoe** [comment]
@darbula looks good. I had but one little comment.

File: django_select2/forms.py (hunk 1 above)
Comment (darbula) [inline django_select2/forms.py]: If I remember right it should be the same in this case. nevertheless here the code is going in line with what is django using in their code in Select widget.

**Comment by codingjoe** [comment]
Released in 5.8.2

**Pull request closed without merging**